password_enhancements.module

```diff
@@ -0,0 +1,471 @@
+  $plugin_config = [];
+  foreach ($constraints as $id => $constraint) {
+    // Replace dot in the constraint's ID with a dash to make it more JS/CSS
+    // friendly.
+    $id = str_replace('.', '-', $constraint->id());
+
+    $type = $constraint->getType();
+
+    /** @var \Drupal\password_enhancements\Plugin\PasswordConstraintPluginInterface $plugin_instance */
+    try {
+      $plugin_instance = $plugin_manager->createInstance($type, $constraint->getConfiguration());
+      $initial_description = $plugin_instance->getInitialDescription();
+    }
+    catch (Exception $e) {
+      Drupal::service('logger.password_enhancements')->exception($e);
+
+      $initial_description = strtr('Failed to create <em>:type</em> plugin instance.', [
+        ':type' => $type,
+      ]);
+
+      // Create an error constraint to show the that this has issues.
+      $id = 'error';
+      $type = 'error';
+      $constraint = new Constraint([
+        'type' => $type,
+        'required' => TRUE,
+        'descriptionSingular' => $initial_description,
+      ], 'password_enhancements_constraints');
+    }
+
+    // Get JS library if set.
+    $plugin_definition = $plugin_manager->getDefinition($type);
+    if (!empty($plugin_definition['jsLibrary'])) {
+      $libraries[] = $plugin_definition['jsLibrary'];
+    }
+    $plugin_config[$type][$id] = $constraint->getConfiguration();
+
+    $arguments = [];
+    foreach ($constraint->getSettings() as $setting => $value) {
+      $arguments["@{$setting}"] = new FormattableMarkup('<span data-setting="@setting">@value</span>', [
+        '@setting' => $setting,
+        '@value' => $value,
+      ]);
+    }
+
+    $processed_constraints[] = [
+      '#theme' => 'password_enhancements_policy_constraint',
+      '#description' => strtr($initial_description, $arguments),
+      '#attributes' => [
+        'id' => $id,
+        'class' => ['constraint'],
+        'data-required' => $constraint->isRequired() ? 'yes' : 'no',
+        'data-validation-passed' => 'no',
+        'data-constraint' => $constraint->getType(),
+      ],
+    ];
+    unset($constraints[$constraint->id()]);
+  }
+
+  // Set wrapper for the constraints.
+  $ajax_wrapper = 'password-policy-constraint-ajax-wrapper';
+  $form['account']['password_policy_constraints'] = [
+    '#weight' => $form['account']['pass']['#weight'] + 0.001,
+    '#prefix' => "<div id='{$ajax_wrapper}'>",
+    '#suffix' => '</div>',
+  ];
+
+  // Try to load policy by its role and priority.
+  try {
+    /** @var \Drupal\password_enhancements\Entity\Storage\PolicyEntityStorageInterface $password_policy_storage */
+    $password_policy_storage = Drupal::entityTypeManager()->getStorage('password_enhancements_policy');
+    $policy = $password_policy_storage->loadByRoleAndPriority($roles);
+  }
+  catch (Exception $e) {
+    Drupal::service('logger.password_enhancements')->exception($e);
+    $policy = NULL;
```

Comment:
This is a typical example for where you should just return early instead of settings the `$policy` to null which requires you to check later whether the `$policy` is an object or null.

Reply:
I have to check either way because `loadByRoleAndPriority()` can return NULL as well.
Here I'm just ensuring it is set to `NULL` to not to run a specific part of the code.

If I would return here then some important part of the code would not run and I can't move the related code before it.